jax/_src/lax/parallel.py

```diff
@@ -730,6 +740,82 @@ def pos_reduce(x):
 core.axis_substitution_rules[pmin_p] = partial(_subst_all_names_in_param, 'axes')
 
 
+# TODO(mattjj): merge with lax.argmin_p / lax.argmax_p
+pargmax_p = core.Primitive('pargmax')
+pargmin_p = core.Primitive('pargmin')
+
+def _pargminmax_impl(prim, x, *, axis):
+  if not isinstance(axis, int): raise NameError(f"unbound axis name: {axis}")
+  return xla.apply_primitive(prim, x, axis=axis)
+pargmax_p.def_impl(partial(_pargminmax_impl, pargmax_p))
+pargmin_p.def_impl(partial(_pargminmax_impl, pargmin_p))
+
+def _pargminmax_abstract_eval(x, *, axis):
+  if isinstance(axis, int):
+    if not 0 <= axis < np.ndim(x):
+      raise ValueError(f"axis index {axis} for operand shape {np.shape(x)}")
+    shape = tuple(np.delete(np.shape(x), axis))
+  else:
+    # TODO(mattjj): avals with names, check and eliminate axis name
+    shape = np.shape(x)
+  return ShapedArray(shape, np.dtype('int32'))
```

Comment:
I thought that `argmin` and `argmax` always return `int64`? Or they do at least when used from the NumPy API. The primitives themselves are parameterized by the integer type.

Reply:
Indeed we'd want that for the `jnp` API, though we don't necessarily need it on the lax API here.

`lax.argmin` and `lax.argmax` are parameterized by the dtype, but because our `lax.axis_index` implementation only returns 32bit values (in turn because XLA's `ReplicaId` produces a `uint32` value) 64bit support here won't be complete. But it's easy to add!